nats.go

```diff
@@ -3790,6 +3790,89 @@ func (m *Msg) RespondMsg(msg *Msg) error {
 	return nc.PublishMsg(msg)
 }
 
+func (m *Msg) isJSMsg() bool {
+	if m.Reply == "" {
+		return false
+	}
+
+	parts := strings.Split(m.Reply, ".")
+	if len(parts) == 8 && parts[0] == "$JS" && parts[1] != "ACK" {
+		return true
+	}
+
+	return false
+}
+
+var (
+	AckAck      = []byte("+ACK")
+	AckNak      = []byte("-NAK")
+	AckProgress = []byte("+WPI")
+	AckNext     = []byte("+NXT")
+	AckTerm     = []byte("+TERM")
+)
+
+// Ack acknowledges a JetStream messages received from a Consumer, indicating the message
+// should not be received again later
+func (m *Msg) Ack() error {
+	if !m.isJSMsg() {
+		return fmt.Errorf("not a JetStream originated message")
+	}
+
+	return m.Respond(AckAck)
+}
+
+// AckConfirm performs an Ack() and requests a confirmation from JetStream that the ack was received
+func (m *Msg) AckConfirm(nc *Conn, timeout time.Duration) (*Msg, error) {
```

Comment:
I think this just waits for the ack to the ack. Does not need to return it etc. This may be a variable option to plain Ack()?

Reply:
Yeah today its a nil message but I think we might later want to include some additional metadata perhaps? Even just a stream name/id for correlation so I didnt want to force a API change if we decided to do that

Reply:
so would you suggest a message signature of `func (m *Msg) Ack(confirm bool, timeout time.Duration) error` or something different?

Reply:
`Ack(opts ...AckOption)`

Reply:
Updated with options, AckNext is different and doesnt really fit that pattern by its nature, so not sure I am a fan of this but for sure it will work and better than adding Ack() and AckConfirm() or something